eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala

```diff
@@ -362,23 +363,156 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
+          stay() sending TxAbort(d.channelId, InvalidRbfFeerate(d.channelId, msg.feerate, minNextFeerate).getMessage)
+        } else {
+          log.info("our peer wants to raise the feerate of the funding transaction (previous={} target={})", d.fundingParams.targetFeerate, msg.feerate)
+          val fundingParams = InteractiveTxParams(
+            d.channelId,
+            d.fundingParams.isInitiator,
+            d.fundingParams.localAmount, // we don't change our funding contribution
+            msg.fundingContribution_opt.getOrElse(d.fundingParams.remoteAmount),
+            d.fundingParams.fundingPubkeyScript,
+            msg.lockTime,
+            d.fundingParams.dustLimit,
+            msg.feerate
+          )
+          val txBuilder = context.spawnAnonymous(InteractiveTxBuilder(
+            remoteNodeId, fundingParams, keyManager,
+            d.commitments.localParams, d.commitments.remoteParams,
+            d.commitments.localCommit.spec.commitTxFeerate,
+            d.commitments.remoteCommit.remotePerCommitmentPoint,
+            d.commitments.channelFlags, d.commitments.channelConfig, d.commitments.channelFeatures,
+            wallet))
+          txBuilder ! InteractiveTxBuilder.Start(self, d.fundingTx +: d.previousFundingTxs.map(_.fundingTx))
+          stay() using d.copy(rbfAttempt = Some(Right(txBuilder))) sending TxAckRbf(d.channelId, fundingParams.localAmount)
+        }
+      }
 
-    case Event(_: TxAckRbf, _: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
-      log.info("rbf not supported yet")
-      stay()
+    case Event(cmd: CMD_BUMP_FUNDING_FEE, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
+      val replyTo = if (cmd.replyTo == ActorRef.noSender) sender() else cmd.replyTo
+      if (!d.fundingParams.isInitiator) {
+        replyTo ! Status.Failure(new RuntimeException("cannot initiate rbf: we're not the initiator of this dual-funding attempt"))
+        stay()
+      } else {
+        d.rbfAttempt match {
+          case Some(_) =>
+            log.warning("cannot initiate rbf, another one is already in progress")
+            replyTo ! Status.Failure(InvalidRbfAlreadyInProgress(d.channelId))
+            stay()
+          case None =>
+            val minNextFeerate = d.fundingParams.targetFeerate * 25 / 24
+            if (cmd.targetFeerate < minNextFeerate) {
+              replyTo ! Status.Failure(InvalidRbfFeerate(d.channelId, cmd.targetFeerate, minNextFeerate))
+              stay()
+            } else {
+              stay() using d.copy(rbfAttempt = Some(Left(cmd.copy(replyTo = replyTo)))) sending TxInitRbf(d.channelId, cmd.lockTime, cmd.targetFeerate, d.fundingParams.localAmount)
+            }
+        }
+      }
+
+    case Event(msg: TxAckRbf, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
+      d.rbfAttempt match {
+        case Some(Left(cmd)) =>
+          log.info("our peer accepted our rbf attempt and will contribute {} to the funding transaction", msg.fundingContribution_opt.getOrElse(0 sat))
```

Comment:
Should `.getOrElse(0 sat)` be done in the attribute definition (which wouldn't return an option)?

Reply:
Good point, done in https://github.com/ACINQ/eclair/pull/2275/commits/ba3baefe8114d5930c8049d530a2224e1b8b104b